agent/agent_endpoint.go

```diff
@@ -86,7 +88,22 @@ func (s *HTTPServer) AgentMetrics(resp http.ResponseWriter, req *http.Request) (
 	if rule != nil && !rule.AgentRead(s.agent.config.NodeName) {
 		return nil, acl.ErrPermissionDenied
 	}
+	if format := req.URL.Query().Get("format"); format == "prometheus" {
+		if s.agent.config.TelemetryPrometheusRetentionTime.Nanoseconds() < 1 {
+			resp.WriteHeader(http.StatusUnsupportedMediaType)
+			fmt.Fprint(resp, "Prometheus is not enable since its retention time is not positive")
+			return nil, nil
+		}
+		handlerOptions := promhttp.HandlerOpts{
+			ErrorLog:           s.agent.logger,
+			ErrorHandling:      promhttp.ContinueOnError,
+			DisableCompression: true,
```

Comment:
Why ? (please add a comment in the code as an answer)

Reply:
Because I was lazy testing it, fixing this in next patch

Reply:
I did simply copy the Nomad way.
In next patch, I removed this boolean, so promhttp now handles it by itself.
Works:
```
curl -H Accept-Encoding:gzip -H "Accept:text/plain; version=0.0.4; charset=utf-8" http://localhost:8500/v1/agent/metrics
```
Returns Prometheus contents GZIPPED